### PR TITLE
feat: show hourly forecast

### PR DESCRIPTION
**Pull request by Ryuno-Ki**
This took me quite some trial and error to figure out.

### CODE REVIEW

**Comment by Ryuno-Ki** [comment]
I want to add an example_response.json to the repo and also respect day/night information as suggested in my last PR before I open this for inclusion.

That being said, by opening a PR now I can invite others to check it out and run locally. Feedback welcome.

File: weather-api-widget/weather.lua [hunk 1]
```diff
@@ -364,6 +366,181 @@ local function worker(user_args)
...
+                if (i - 1) % 5 == 0 then
+                    table.insert(hours_below, wibox.widget {
+                        text = os.date(time_format_12h and '%I%p' or '%H:00', tonumber(hour.time_epoch)),
+                        align = 'center',
+                        font = font_name .. ' 9',
+                        widget = wibox.widget.textbox
+                    })
+
+                    table.insert(temp_below, wibox.widget {
+                        markup = '<span foreground=""'
```
Comment (trap000d) [inline weather-api-widget/weather.lua:442]: Typo? Extra double quote results syntax error
 ```
 markup = '<span foreground="'
```

This works.

Comment (Ryuno-Ki) [reply weather-api-widget/weather.lua:442]: Overly clever NeoVim, but yes.

**Comment by trap000d** [comment]
Hi @Ryuno-Ki,

Would you consider to [optionally] add popup on mouse hover?

```lua
    weather_widget:connect_signal("mouse::enter", function()
        weather_widget:set_bg(beautiful.bg_focus)
        weather_popup:move_next_to(mouse.current_widget_geometry)
    end)

    weather_widget:connect_signal("mouse::leave", function()
        if weather_popup.visible then
            weather_widget:set_bg('#00000000')
            weather_popup.visible = not weather_popup.visible
        end
    end)
```

**Comment by Ryuno-Ki** [comment]
Done.

**Comment by Ryuno-Ki** [comment]
I _think_ I have everythink now. I can see unrelated failures in luacheck to this extension.

@streetturtle Mind taking a review on this?

**Comment by trap000d** [comment]
All works great, thanks a lot.
The only issue I could spot: all night icons files are missing.
You might want to add all them to corresponding directories.

**Comment by Ryuno-Ki** [comment]
Good catch!

**Review by streetturtle — APPROVED** [review]

**Comment by streetturtle** [comment]
This is great! Thank you! :+1:

**Pull request merged**